docs/reference/architecture.md

```diff
@@ -1 +1,52 @@
-# Architecture 
+# Architecture
+
+This section provides an overview of the software architecture as well as the
+Docker setup.
+
+## Overview :bookmark_tabs:
+
+This overview should complement the [data flow](../data-flow.md) description and
+goes into more technical detail. The Django 4[^1] server combines all components
+into a single REST-API, which is then used by the VueJs[^2] frontend. Django uses
+gRPC with Protocol Buffers[^3] to query SEEREP[^4] for scalar data. Protocol
+Buffers is a language and platform neutral mechanism for serializing data,
+developed and open-sourced by Google. It's equivalent to JSON, except it's more
+efficient and much faster.
+
+Downloading the point clouds from SEEREP every time a request for them is passed
+to the Django server would take too long for a satisfactory experience, so they
+are cached locally. For that, the RabbitMQ[^5] message broker handles
+asynchronous tasks to download the point clouds in predetermined time intervals.
```

Comment:
```suggestion
are cached locally. For that, Celery[^9] handles asynchronous tasks using the RabbitMQ[^5]
message broker to download the point clouds in predetermined time intervals.
```

Since #161 is still open, I don't think we download them in time intervals. They are just downloaded on demand.